[ASCII-2054] Adding check presence of Agent check's metadata in datadog-agent status command output

misteriaud opened the pull request:
<!--
* Contributors are encouraged to read our [CONTRIBUTING](/CONTRIBUTING.md) documentation.
* Both Contributor and Reviewer Checklists are available at https://datadoghq.dev/datadog-agent/guidelines/contributing/#pull-requests.
* The pull request:
  * Should only fix one issue or add one feature at a time.
  * Must update the test suite for the relevant functionality.
  * Should pass all status checks before being reviewed or merged.
* Commit titles should be prefixed with general area of pull request's change.
* Please fill the below sections if possible with relevant information or links.
-->
### What does this PR do ?
This PR adds check presence of Agent check's metadata in `datadog-agent status`command output.

### Motivation
Agent check's metadata was missing between version `7.51` and `7.56`.
This regression was fixed in this PR: #27801

### Describe how to test/QA your changes
This PR only update an e2e test.

misteriaud committed:
test the presence of metadata checks output in status subcommand e2e test

pgimalac commented on test/new-e2e/tests/agent-subcommands/status/status_common_test.go:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏Seems a little weird to me that `TestDefaultInstallStatus` checks for the presence of a non-default line, and that it requires a non-default configuration 😄 
What do you think of adding a separate common test which would configure the agent with the custom check and ensure it contains those strings ?

<!-- Warning represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**🔨 warning:** ‏Also you add the files in `e2e.Run` but IMO they should only be added for the specific test which needs it with `v.UpdateEnv` (and then removed) to avoid "polluting" the config for other tests, as that could lead to unexpected behavior

misteriaud committed:
moving metadata check in its own test function

misteriaud committed:
Fix mistake

I forget to remove custom_check provisioning at the test root instead of putting it only in the related test

misteriaud committed:
adding missing WithOS() directive in windows test updateEnv

misteriaud committed:
refactor e2e test for status subcommand

misteriaud committed:
refactor fetchAndCheckStatus function for easier code review

misteriaud commented:
/merge
